server/shared/src/main/scala/org/http4s/server/middleware/Throttle.scala

```diff
@@ -74,11 +74,12 @@ object Throttle {
                   val attemptSet: F[Option[TokenAvailability]] =
                     if (newTokenTotal >= 1)
                       setter((newTokenTotal - 1, currentTime))
-                        .map(_.guard[Option].as(TokenAvailable))
+                        .map(Option.when(_)(TokenAvailable))
```

Comment:
Yeah, I like `Option.when` and `Option.unless` too. Unfortunately they don't exist for Scala 2.15.

Reply:
Oh, that's a shame... Well, removing...

Reply:
Darn. Sorry for interrupting.